pkg/provider/eks/eks.go

```diff
@@ -0,0 +1,582 @@
+		if aerr, ok := err.(awserr.Error); ok && aerr.Code() == eks.ErrCodeResourceNotFoundException {
+			return true, nil
+		}
+		return false, fmt.Errorf("Couldn't get nodegroupname status: %v", err)
+	}
+
+	log.Printf("Nodegroup '%v' for Cluster '%v' status: %v", nodegroupName, clusterName, *nodegroupRes.Nodegroup.Status)
+	return false, nil
+}
+
+// AllNodeGroupsRunning returns an error if at least one node pool is not running
+func (c *EKS) AllNodeGroupsRunning(*kingpin.ParseContext) error {
+	req := &eksCluster{}
+	for _, deployment := range c.eksResources {
+		if err := yamlGo.UnmarshalStrict(deployment.Content, req); err != nil {
+			return fmt.Errorf("Error parsing the cluster deployment file %s:%v", deployment.FileName, err)
+		}
+		for _, nodegroup := range req.NodeGroups {
+			isRunning, err := c.nodeGroupCreated(*nodegroup.NodegroupName, *req.Cluster.Name)
+			if err != nil {
+				return fmt.Errorf("error fetching nodegroup info")
+			}
+			if !isRunning {
+				return fmt.Errorf("nodepool not running name: %v", *nodegroup.NodegroupName)
+			}
+		}
+	}
+	return nil
+}
+
+// AllNodeGroupsDeleted returns an error if at least one node pool is not deleted
+func (c *EKS) AllNodeGroupsDeleted(*kingpin.ParseContext) error {
+	req := &eksCluster{}
+	for _, deployment := range c.eksResources {
+		if err := yamlGo.UnmarshalStrict(deployment.Content, req); err != nil {
+			return fmt.Errorf("Error parsing the cluster deployment file %s:%v", deployment.FileName, err)
+		}
+		for _, nodegroup := range req.NodeGroups {
+			isRunning, err := c.nodeGroupDeleted(*nodegroup.NodegroupName, *req.Cluster.Name)
+			if err != nil {
+				return fmt.Errorf("error fetching nodegroup info")
+			}
+			if !isRunning {
+				return fmt.Errorf("nodepool not running name: %v", *nodegroup.NodegroupName)
+			}
+		}
+	}
+	return nil
+}
+
+// NewK8sProvider sets the k8s provider used for deploying k8s manifests
+func (c *EKS) NewK8sProvider(*kingpin.ParseContext) error {
+
+	clusterName := c.DeploymentVars["CLUSTER_NAME"]
+	region := c.DeploymentVars["ZONE"]
+
+	req := &eks.DescribeClusterInput{
+		Name: &clusterName,
+	}
+
+	rep, err := c.clientEKS.DescribeCluster(req)
+	if err != nil {
+		return fmt.Errorf("failed to get cluster details: %v", err)
+	}
+
+	arnRole := *rep.Cluster.Arn
+
+	caCert, err := base64.StdEncoding.DecodeString(*rep.Cluster.CertificateAuthority.Data)
+	if err != nil {
+		return fmt.Errorf("failed to decode certificate: %v", err.Error())
+	}
+
+	cluster := clientcmdapi.NewCluster()
+	cluster.CertificateAuthorityData = []byte(caCert)
+	cluster.Server = *rep.Cluster.Endpoint
+
+	clusterContext := clientcmdapi.NewContext()
+	clusterContext.Cluster = arnRole
+	clusterContext.AuthInfo = arnRole
+
+	authInfo := clientcmdapi.NewAuthInfo()
+	authInfo.Exec = &clientcmdapi.ExecConfig{
+		APIVersion: "client.authentication.k8s.io/v1alpha1",
+		Command:    "aws",
+		Args:       []string{"--region", region, "eks", "get-token", "--cluster-name", clusterName},
```

Comment:
Do we need the aws cli installed for running this by any chance? no right?

Reply:
No , I don't think so.